soundata/datasets/tut_sound_events_2017.py

```diff
@@ -0,0 +1,431 @@
+
+    @property
+    def audio(self) -> Optional[Tuple[np.ndarray, float]]:
+        """The clip's audio
+
+        Returns:
+            * np.ndarray - audio signal
+            * float - sample rate
+
+        """
+        return load_audio(self.audio_path)
+
+    @property
+    def split(self):
+        return self._clip_metadata.get("split")
+
+    @core.cached_property
+    def events(self) -> Optional[annotations.Events]:
+        return load_events(self.annotations_path)
+
+    @core.cached_property
+    def non_verified_events(self) -> Optional[annotations.Events]:
+        return load_events(self.non_verified_annotations_path)
+
+    def to_jams(self):
+        """Get the clip's data in jams format
+
+        Returns:
+            jams.JAMS: the clip's data in jams format
+
+        """
+        # raise NotImplementedError
+
+        # TODO: add events
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path,
+            metadata=self._metadata(),
+        )
+
+
+@io.coerce_to_bytes_io
+def load_audio(fhandle: BinaryIO, sr=None) -> Tuple[np.ndarray, float]:
+    """Load a TUT Sound events 2017 audio file.
+
+    Args:
+        fhandle (str or file-like): File-like object or path to audio file
+        sr (int or None): sample rate for loaded audio, None by default, which
+            uses the file's original sample rate of 44100 without resampling.
+
+    Returns:
+        * np.ndarray - the mono audio signal
+        * float - The sample rate of the audio file
+
+    """
+    audio, sr = librosa.load(fhandle, sr=sr, mono=False)
+    return audio, sr
+
+
+@io.coerce_to_string_io
+def load_events(fhandle: TextIO) -> annotations.Events:
+    """Load an TUT Sound events 2017 annotation file
+    Args:
+        fhandle (str or file-like): File-like object or path to the sound
+        events annotation file
+
+    Returns:
+        Events: sound events annotation data
+    """
+
+    times = []
+    labels = []
+    confidence = []
+    reader = csv.reader(fhandle, delimiter="\t")
+    for line in reader:
+        offset = (
+            0 if len(line) == 3 else 2
+        )  # ann files in dev and eval have different format
+        times.append([float(line[offset]), float(line[offset + 1])])
+        labels.append(line[offset + 2])
+        confidence.append(1.0)
+
+    events_data = annotations.Events(np.array(times), labels, np.array(confidence))
+    return events_data
+
+
+@core.docstring_inherit(core.Dataset)
+class Dataset(core.Dataset):
+    """
+    The TUT Sound events 2017 dataset
+    """
+
+    def __init__(self, data_home=None):
+        super().__init__(
+            data_home,
+            name="tut_sound_events_2017",
+            clip_class=Clip,
+            bibtex=BIBTEX,
+            remotes=REMOTES,
+            license_info=LICENSE_INFO,
+        )
+
+    @core.copy_docs(load_audio)
+    def load_audio(self, *args, **kwargs):
+        return load_audio(*args, **kwargs)
```

Comment:
The idea of exposing the functions at the dataset level is for the user to be able to use the load functions directly if needed. For that to be complete add here the `load_events` the same way you added the `load_audio`